dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/ApplicationProcessor.java

```diff
@@ -0,0 +1,234 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.container.service.app.deploy.ApplicationInfo;
+import com.ibm.ws.container.service.app.deploy.WebModuleInfo;
+import com.ibm.ws.ffdc.annotation.FFDCIgnore;
+import com.ibm.ws.microprofile.openapi.impl.core.util.Json;
+import com.ibm.ws.microprofile.openapi.impl.core.util.Yaml;
+import com.ibm.ws.microprofile.openapi.impl.jaxrs2.Reader;
+import com.ibm.ws.microprofile.openapi.impl.model.OpenAPIImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.PathsImpl;
+import com.ibm.ws.microprofile.openapi.impl.model.info.InfoImpl;
+import com.ibm.ws.microprofile.openapi.impl.parser.OpenAPIV3Parser;
+import com.ibm.ws.microprofile.openapi.impl.parser.core.models.SwaggerParseResult;
+import com.ibm.ws.microprofile.openapi.utils.OpenAPIModelWalker;
+import com.ibm.ws.microprofile.openapi.utils.OpenAPIUtils;
+import com.ibm.ws.microprofile.openapi.utils.ServerInfo;
+import com.ibm.wsspi.adaptable.module.Container;
+import com.ibm.wsspi.http.VirtualHost;
+
+/**
+ *
+ */
+@Component(service = { ApplicationProcessor.class }, configurationPolicy = ConfigurationPolicy.IGNORE, immediate = true, property = { "service.vendor=IBM" })
+public class ApplicationProcessor {
+
+    private static final TraceComponent tc = Tr.register(ApplicationProcessor.class);
+
+    enum DocType {
+        JSON,
+        YAML
+    }
+
+    private OpenAPI document = null;
+    private ApplicationInfo currentApp = null;
+    private final ServerInfo serverInfo = new ServerInfo();
+
+    public void activate(ComponentContext cc) {
+        this.document = createBaseOpenAPIDocument();
+    }
+
+    public void processApplication(ApplicationInfo appInfo) {
+        synchronized (this.document) {
+
+            if (currentApp != null) {
+                return;
+            }
+
+            if (appInfo == null) {
+                return;
+            }
+
+            Container appContainer = appInfo.getContainer();
+            if (appContainer == null) {
+                return;
+            }
+
+            WebModuleInfo moduleInfo = OpenAPIUtils.getWebModuleInfo(appContainer);
+            if (moduleInfo == null) {
+                return;
+            }
+            ClassLoader appClassloader = moduleInfo.getClassLoader();
+            String contextRoot = moduleInfo.getContextRoot();
+
+            boolean isOASApp = false;
+
+            //read and process the MicroProfile config
+            ConfigProcessor configProcessor = new ConfigProcessor(appClassloader);
+
+            OpenAPI newDocument = null;
+            //Retrieve model from model reader
+            OASModelReader modelReader = OpenAPIUtils.getOASModelReader(appClassloader, configProcessor.getModelReaderClassName());
+            if (modelReader != null) {
+                try {
+                    new OASFactoryResolverImpl();
```

Comment:
Line `101` looks strange.  Do we really need this before using the model, or is this to cover a time-window where we may have not initialized the resolver yet?